src/assertions/OrderAssertions.ts

```diff
@@ -173,15 +183,78 @@ export class OrderAssertions {
     // All 0x signed fill order fillAmounts are filled using the makerTokenAmount of the
     // signedIssuanceOrder so we need to make sure that signedIssuanceOrder.makerTokenAmount
     // has enough for the 0x orders (scaled by fraction of order quantity being filled).
-    const {
-      makerTokenAmount,
-      quantity,
-    } = signedIssuanceOrder;
     this.commonAssertions.isGreaterOrEqualThan(
       signedIssuanceOrder.makerTokenAmount.mul(quantityToFill).div(quantity),
       makerTokensUsed,
       coreAPIErrors.MAKER_TOKEN_INSUFFICIENT(signedIssuanceOrder.makerTokenAmount, makerTokensUsed),
     );
+
+    // Ensure that the liquidity orders that we have specified can fill the amount of the
+    // issuance order that we're trying to fill.
+    this.isValidLiquidityAmounts(
+      signedIssuanceOrder,
+      quantityToFill,
+      requiredComponentFills,
+    );
+  }
+
+  /* ============ Private Helpers =============== */
+
+  private isValidLiquidityAmounts(
+    signedIssuanceOrder: SignedIssuanceOrder,
+    quantityToFill: BigNumber,
+    requiredComponentFills: { [addr: string]: BigNumber },
+  ) {
+    const {
+      quantity,
+      requiredComponents,
+      requiredComponentAmounts,
+    } = signedIssuanceOrder;
+
+    _.each(requiredComponents, (component, i) => {
+      const issuanceOrderComponentPartialAmount =
+        calculatePartialAmount(requiredComponentAmounts[i], quantityToFill, quantity);
+
+      this.commonAssertions.isEqual(
```

Comment:
Instead of `isEqual`, create a new CommonAssertion that is `isEqualBigNumber` or so that compares two big numbers.